core/src/traits/Kokkos_WorkTagTrait.hpp

```diff
@@ -110,6 +110,9 @@ struct WorkTagTrait : TraitSpecificationBase<WorkTagTrait> {
   template <class T>
   using trait_matches_specification = std::integral_constant<
       bool, !std::is_void<T>::value &&
+#ifndef KOKKOS_ENABLE_DEPRECATED_CODE_3
+                std::is_empty<T>::value &&
```

Comment:
When the deprecation is removed, we should get rid of the `!std::is_void<T>::value` check, correct?  Maybe leave a comment to that effect, or a `#else` on the macro?

Reply:
No, when the deprecation is removed we drop the macro guard that is all.
`void` is not a valid `WorkTag`

Reply:
Gotcha.  Will change.